CRM/Contribute/BAO/Contribution.php

```diff
@@ -3858,7 +3858,86 @@ public static function recordAdditionalPayment($contributionId, $trxnsData, $pay
     $params['skipLineItem'] = TRUE;
     $trxnsData['trxn_date'] = !empty($trxnsData['trxn_date']) ? $trxnsData['trxn_date'] : date('YmdHis');
     $arAccountId = CRM_Contribute_PseudoConstant::getRelationalFinancialAccount($contributionDAO->financial_type_id, 'Accounts Receivable Account is');
-    if ($paymentType == 'owed') {
+    $fiancialItemPaidStatusID = CRM_Core_PseudoConstant::getKey('CRM_Financial_DAO_FinancialItem', 'status_id', 'Paid');
+    if ($paymentType == 'overpaid') {
```

Comment:
$paymentType as variable name for 'owed' or 'refund' makes some sense. 'overpaid' seems not quite right. Let's change to 'owed', 'more than owed', and 'refund'.